aten/src/THNN/generic/pooling_shape.h

```diff
@@ -14,4 +14,17 @@ static inline T pooling_output_shape(
     return outputSize;
 }
 
+template<typename T>
+static inline T pooling_output_shape_pad_lr(
```

Comment:
Could you also change the `pooling_output_shape` above to use this new `pooling_output_shape_pad_lr` (by passing pad as both pad_l and pad_r), so that we don't have (almost) duplicated implementation.